src/clients/FtpGateway.cc

```diff
@@ -2579,7 +2579,7 @@ Ftp::Gateway::haveParsedReplyHeaders()
     // makePublic() if allowed/possible or release() otherwise
     if (flags.authenticated || // authenticated requests can't be cached
             getCurrentOffset() ||
-            !e->makePublic()) {
+            !e->makePublic(false)) {
         e->release();
```

Comment:
I assume that in case of a failed makePublic() call failure, this `release()` call does nothing. Please correct me if that assumption is wrong.

P.S. I am aware that this `release()` call is needed for the other two `if` statement conditions.